src/models/OAuthModel.php

```diff
@@ -125,16 +125,22 @@ public function createAccessTokenFromPassword($clientId, $username, $password)
      */
     protected function getUserId($username, $password)
     {
-        $sql  = 'SELECT ID, password, email FROM user
-            WHERE username=:username
-            AND verified = 1';
+        $sql  = 'SELECT ID, password, email, verified FROM user
+            WHERE username=:username';
         $stmt = $this->_db->prepare($sql);
         $stmt->execute(array("username" => $username));
         $result = $stmt->fetch(PDO::FETCH_ASSOC);
+
         if ($result) {
-            if (password_verify(md5($password), $result['password'])) {
```

Comment:
Yeah. Along that way.

I'd even say ```if (! password_verify …  return false;``` (or ```throw Exception```?

I always try to return early in error conditions and have the "expected" outcome right at the end of the function.